apps/console/src/features/email-templates/pages/email-template-types-page.tsx

```diff
@@ -146,8 +166,52 @@ const EmailTemplateTypesPage: FunctionComponent<EmailTemplateTypesPagePropsInter
      * @param {number} offset - Pagination offset value.
      * @param {number} limit - Pagination item limit.
      */
-    const paginate = (list: EmailTemplateType[], offset: number, limit: number): void => {
-        setPaginatedEmailTemplateTypes(list.slice(offset, limit + offset));
+    const paginate = (list: EmailTemplateType[], offset: number, limit: number): EmailTemplateType[] => {
+        return list.slice(offset, limit + offset);
+    };
+
+    const handleSearch = (query: string): void => {
+        try {
+            // TODO: Implement using API once the API is ready
+            setFilteredEmailTemplateTypes(filterList(emailTemplateTypes, query, "displayName", true));
+            setSearchQuery(query);
+            setListOffset(0);
+            setResetPagination();
+        } catch (error) {
+            dispatch(addAlert({
+                description: error.message,
+                level: AlertLevels.ERROR,
+                message: t("adminPortal:components.userstores.advancedSearch.error")
+            }));
+        }
+    }
+
+    /**
+     * Handles the `onSearchQueryClear` callback action.
+     */
+    const handleSearchQueryClear = (): void => {
+        setTriggerClearQuery(!triggerClearQuery);
+        setSearchQuery("");
+        setFilteredEmailTemplateTypes(emailTemplateTypes);
+    };
+
+    /**
+     * Handle sort strategy change.
+     *
+     * @param {React.SyntheticEvent<HTMLElement>} event.
+     * @param {DropdownProps} data.
+     */
+    const handleSortStrategyChange = (event: React.SyntheticEvent<HTMLElement>, data: DropdownProps) => {
```

Comment:
Missing return type. Please check other places as well.

```suggestion
    const handleSortStrategyChange = (event: React.SyntheticEvent<HTMLElement>, data: DropdownProps): void => {
```

Reply:
Resolved by 7990967